go/cmd/zk/zkcmd.go

```diff
@@ -134,14 +134,16 @@ func main() {
 	defer exit.Recover()
 	defer logutil.Flush()
 	server = pflag.String("server", "", "server(s) to connect to")
-
-	// handling case of --help, -help & -h
+	// handling case of --help & -h
 	args := os.Args[1:]
 	var help bool
-	if args[0] == "-h" {
-		pflag.BoolVarP(&help, "help", "h", false, "display usage and exit")
-	} else if args[0] == "--help" || args[0] == "-help" {
-		pflag.BoolVar(&help, "help", false, "display usage and exit")
+	pflag.BoolVarP(&help, "help", "h", false, "display usage and exit")
+	log.RegisterFlags(pflag.CommandLine)
+	logutil.RegisterFlags(pflag.CommandLine)
+
+	pflag.CommandLine.Usage = func() {
+		fmt.Fprintf(os.Stderr, doc)
+		pflag.Usage()
```

Comment:
pretty sure this creates a stack overflow, have you tested this?

Reply:
No pflg.usage and pflag.commandline.Usage are different functions.. I initially thought same but it isn't the case